pkg/oc/admin/prune/imageprune/prune.go

```diff
@@ -56,8 +54,24 @@ const (
 	// ReferencedImageLayerEdgeKind defines an edge from an ImageStreamNode or an
 	// ImageNode to an ImageComponentNode.
 	ReferencedImageLayerEdgeKind = "ReferencedImageLayer"
+
+	// ReferencedImageManifestEdgeKind defines an edge from an ImageStreamNode or an
+	// ImageNode to an ImageComponentNode.
+	ReferencedImageManifestEdgeKind = "ReferencedImageManifest"
+
+	// UnreferencedImageComponentEdgeKind is an edge from an ImageNode to an ImageComponentNode denoting that
+	// the component is currently being unreferenced in a running job.
+	UnreferencedImageComponentEdgeKind = "UnreferencedImageComponentToDelete"
```

Comment:
This feels very artificial. Why do we need these "negative" references?

Can we structure the pruner such a way that we remove only nodes which don't have any references?

Reply:
+1 on tracking this elsewhere.

Reply:
Reworked and simplified a bit.